src/Microsoft.ML.CpuMath/CpuMathUtils.DotNetCoreApp.cs

```diff
@@ -0,0 +1,362 @@
+        {
+            Contracts.AssertNonEmpty(src);
+            Contracts.Assert(0 < count && count <= src.Length);
+            Contracts.AssertNonEmpty(dst);
+            Contracts.Assert(count <= dst.Length);
+
+            if (Sse.IsSupported)
+            {
+                SseIntrinsics.AddU(src, dst, count);
+            }
+            else
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    dst[i] += src[i];
+                }
+            }
+        }
+
+        public static void Add(float[] src, int[] indices, float[] dst, int count)
+        {
+            Contracts.AssertNonEmpty(src);
+            Contracts.Assert(0 < count && count <= src.Length);
+            Contracts.AssertNonEmpty(indices);
+            Contracts.Assert(count <= indices.Length);
+            Contracts.AssertNonEmpty(dst);
+            Contracts.Assert(count < dst.Length);
+
+            if (Sse.IsSupported)
+            {
+                SseIntrinsics.AddSU(src, indices, dst, count);
+            }
+            else
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    dst[indices[i]] += src[i];
+                }
+            }
+        }
+
+        public static void Add(float[] src, int[] indices, float[] dst, int dstOffset, int count)
+        {
+            Contracts.AssertNonEmpty(src);
+            Contracts.Assert(0 < count && count <= src.Length);
+            Contracts.AssertNonEmpty(indices);
+            Contracts.Assert(count <= indices.Length);
+            Contracts.AssertNonEmpty(dst);
+            Contracts.Assert(0 <= dstOffset && dstOffset < dst.Length);
+            Contracts.Assert(count <= dst.Length - dstOffset);
+
+            ArraySegment<float> dstSeg = new ArraySegment<float>(dst, dstOffset, count);
+            Add(src, indices, dstSeg.Array, count);
+        }
+
+        public static void MulElementWise(float[] src1, float[] src2, float[] dst, int count)
+        {
+            Contracts.AssertNonEmpty(src1);
+            Contracts.Assert(0 < count && count <= src1.Length);
+            Contracts.AssertNonEmpty(src2);
+            Contracts.Assert(0 < count && count <= src2.Length);
+            Contracts.AssertNonEmpty(dst);
+
+            if (Sse.IsSupported)
+            {
+                SseIntrinsics.MulElementWiseU(src1, src2, dst, count);
+            }
+            else
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    dst[i] = src1[i] * src2[i];
+                }
+            }
+        }
+
+        public static float SumSq(float[] src, int count)
+        {
+            Contracts.AssertNonEmpty(src);
+            Contracts.Assert(0 < count && count <= src.Length);
+
+            if (Sse.IsSupported)
+            {
+                return SseIntrinsics.SumSqU(src, count);
+            }
+            else
+            {
+                // Software fallback.
```

Comment:
It seems odd to have this comment in just one method.  I think it can be removed as it doesn't provide much value IMO.